runway/hooks/staticsite/upload_staticsite_env.py

```diff
@@ -0,0 +1,70 @@
+"""CFNgin hook for uploading environment files to a static website S3 bucket."""
```

Comment:
does this need its own file? it seems like it could be a new function in `upload_staticsite.py`.

also, shouldn't this be performing some kind of hash calculation/cache invalidation so that the cloudfront distribution is updated when these files change?

add documentation for this new hook to [this page](https://docs.onica.com/projects/runway/en/release/cfngin/hooks.html).

Reply:
I added in support to invalidate the cache. This doesn't do any kind of hash checking though. Basically, any time a file is uploaded is the trigger for when the cache is invalidated. I think we can optimize invalidation later in a different PR.